src/twyn/dependency_parser/abstract_parser.py

```diff
@@ -1,43 +1,29 @@
 import logging
-import os
 from abc import ABC, abstractmethod
-from pathlib import Path
 
-from twyn.base.exceptions import TwynError
-from twyn.dependency_parser.exceptions import (
-    PathIsNotFileError,
-    PathNotFoundError,
-)
+from twyn.file_handler.file_handler import BaseFileHandler, FileHandlerPathlib
 
 logger = logging.getLogger()
 
 
 class AbstractParser(ABC):
-    def __init__(self, file_path: str = "") -> None:
-        self.file_path = Path(os.path.abspath(os.path.join(os.getcwd(), file_path)))
+    """
+    Abstract class for file parsers.
 
-    def __str__(self):
+    Provides basic methods to deal with the dependecies file.
+    """
+
+    def __init__(self, file_path: str, file_handler: BaseFileHandler = FileHandlerPathlib) -> None:
```

Comment:
```suggestion
    def __init__(self, file_path: str, file_handler: BaseFileHandler = type[FileHandlerPathlib]) -> None:
```

Reply:
Nice catch!

I was in doubt about receiving the FileHandler object directly or just the type though. For now I think it's fine, as long as we don't have another FileHandler type that requires some other parameters to initialize

Reply:
Usually, I'd say it's best to pass instances instead of classes. Unless you have a very good reason to defer the class instantiation.

Reply:
After checking, I think it's best to just not make AbstractParser receive any FileHandler and assume it's going to use FileHandlerPathlib.

If we wanted to make it receive an object, DependencySelector should tell the AbstractParser which FileHandler to use, but right now I think it's safe to say that we only want one way of opening/reading files. I'd leave it like this, tying AbstractParser to FileHandlerPathlib, and see how it evolves if we agree :)